minima/database.nim

```diff
@@ -20,38 +20,55 @@ type
         KeyNotFound             = "minima: key not found"
         PersistenceFailed       = "minima: persistence failed"
 
-proc log(db: Database, key: seq[byte], value: seq[byte]) =
-    let write = concat(
-        @(uint32(len(key)).toBytes),
-        @(uint32(len(value)).toBytes),
-        key,
-        value
-    )
-
-    # @TODO CATCH EXCEPTIONS
-    discard db.log.writeBytes(write, 0, len(write))
-    db.log.flushFile()
+proc toAESKey*(str: string): array[aes256.sizeKey, byte] =
+    ## Concerts a string to an AES Key for opening an encrypted database.
```

Comment:
In general, https://nim-lang.org/docs/nep1.html#introduction-spacing-and-whitespace-conventions convention is to use 2-space indentation. A bit trivial/cosmetic, but does seem to be consistent within Nim codebases.